src/util-var.h

```diff
@@ -48,15 +48,15 @@ enum VarTypes {
 
 /** \todo see ticket #6855. The type field should be 16 bits. */
 typedef struct GenericVar_ {
-    uint8_t type; /**< variable type, uses detection sm_type */
-    uint8_t pad[3];
```

Comment:
Could you please tell the reasoning for this?
My understanding is that this struct is there to create a common data type for different parts of the code. Given that the only place where this padding is used is `flowvar`, I'm unable to understand why does it even have to be 2 bytes.. It seems to me that just a `u8` can suffice.

Reply:
`GenericVar` represents a `FlowVar` and `FlowBit` so the `type`, `idx` and `next` members of each has to align

Reply:
oh. Do you mean they should be cacheline aligned? So, you're trying to avoid a one byte hole and keeping it 32B + 32B?

Reply:
I'm using `uint8_t pad[2]` to avoid the hole. With either solution -- a single or double `pad`, the result is the same because the following item is a 32-bit item.